File: C++/I/07-CURSORS/1/cursorbutton.cpp
#include "cursorbutton.h"
#include "form.h"

CursorButton::CursorButton(QWidget *parent):
   QPushButton(parent){}

void CursorButton::setTag(int tag)
{
    this->_tag = tag;
}

int CursorButton::tag()
{
    return _tag;
}

void CursorButton::mousePressEvent(QMouseEvent *event)
{
    Form *p = (Form*)parent()->parent();
    int c = p->names()->count();
    switch (event->button())
    {
        case Qt::LeftButton:
            _tag = (_tag + 1) % c;
            break;
        case Qt::RightButton:
            _tag = (_tag - 1 + c) % c;
            break;
        default:
            break;
    }
    setText(p->names()->at(_tag));
    setCursor(p->cursors()->at(_tag));
}

File: Python/I/07-CURSORS/2/cursorbutton.py
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *

class CursorButton(QPushButton):
    mousePressed = pyqtSignal(QMouseEvent)
    
    def mousePressEvent(self, event):
        self.mousePressed.emit(event)
